wes_qc/hail_utils.py

```diff
@@ -18,19 +20,22 @@ def clear_temp_folder(tmp_dir: str) -> None:
     os.makedirs(tmp_dir, exist_ok=True)
 
 
-def init_hl(tmp_dir: str) -> pyspark.SparkContext:
+def init_hl(tmp_dir: str) -> Optional[pyspark.SparkContext]:
     clear_temp_folder(tmp_dir)
-    print("=== Checking for SparkContext ===")
-    sc = pyspark.SparkContext.getOrCreate()
 
+    # The latest Hail version (0.2.133) automatically catches spark both for local run and for cluster
+    # We don't need to call for SPARK context manually
+    # print("=== Probing for SparkContext ===")
+    # sc = pyspark.SparkContext.getOrCreate()
     # sc = pyspark.SparkContext()
+    # hl.init(sc=sc, tmp_dir=tmp_dir, idempotent=True)
+
     print("=== Initializing Hail ===")
-    hl.init(sc=sc, tmp_dir=tmp_dir, idempotent=True)
+    hl.init(tmp_dir=tmp_dir, idempotent=True)
     hl.default_reference("GRCh38")
-    return sc
 
 
-def stop_hl(sc: pyspark.SparkContext) -> None:
+def stop_hl(sc: Optional[pyspark.SparkContext] = None) -> None:
```

Comment:
this is probably unnecessary

Reply:
Commented this for now. After a while, when we are sure that Hail init works correctly in all cases, we can delete it.